source/write/transaction.txt

```diff
@@ -0,0 +1,183 @@
+       |
+       | You can set transaction options by passing an options parameter.
+
+   * - ``commitTransaction()``
+     - | Commits the active transaction for this session. This method returns an
+         error if there is no active transaction for the session or the
+         transaction was previously ended.
+
+   * - ``abortTransaction()``
+     - | Ends the active transaction for this session. This method returns an
+         error if there is no active transaction for the session or if the
+         transaction was committed or ended.
+
+.. _php-txn-example:
+
+Transaction Example
+-------------------
+
+This example defines a callback function that
+modifies data in the collections of the ``bank`` database for a
+banking transaction. The code performs the following actions:
+
+- Creates ``Collection`` instances to access the target
+  collections.
+- Specifies the account number and amount to be transferred between
+  accounts.
+- Defines the callback function, passing the ``Session`` instance as a
+  parameter.
+- Updates the customer's balances to reflect the money transfer.
+- Records a receipt of the transaction with a timestamp.
+- Prints a message if the transaction committed successfully.
+
+.. literalinclude:: /includes/write/transaction.php
+   :copyable:
+   :language: php
+   :dedent:
+   :start-after: begin-callback
+   :end-before: end-callback
+
+Then, run the following code to perform the transaction. This code
+completes the following actions:
+
+1. Creates a session from the client by using the ``startSession()`` method.
+#. Calls the ``with_transaction()`` method to manage the transaction,
+   passing the session and the callback as parameters.
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/write/transaction.php
+      :language: php
+      :dedent:
+      :start-after: begin-txn
+      :end-before: end-txn
+      :emphasize-lines: 1, 4
+
+   .. output:: 
+      :language: console
+      :visible: false
+
+      Successfully performed transaction!
+      Summary: SAVINGS +1000 CHECKING -1000
+
+Additional Information
+----------------------
+
+To learn more about the concepts mentioned in this guide, see the
+following pages in the {+mdb-server+} manual:
+
+- :manual:`Transactions </core/transactions/>`
+- :manual:`Server Sessions </reference/server-sessions/>`
+- :manual:`Read Isolation, Consistency, and Recency
+  </core/read-isolation-consistency-recency/>`
+
+To learn more about ACID compliance, see the :website:`What are ACID
+Properties in Database Management Systems? </basics/acid-transactions>`
+article on the MongoDB website.
+
+To learn more about insert operations, see the
+:ref:`php-write-insert` guide.
+
+API Documentation
+~~~~~~~~~~~~~~~~~
+
+To learn more about the methods and types mentioned in this
+guide, see the following API documentation:
+
+- :phpclass:`MongoDB\Client`
+- :phpmethod:`MongoDB\Client::startSession()`
+- :phpmethod:`MongoDB\with_transaction()`
```

Comment:
I'll note that the docs use a common `:phpmethod:` directive here, even though this is technically a function. Nothing to address here -- just wanted to point that out.